src/databricks/sql/client.py

```diff
@@ -297,6 +298,58 @@ def _check_not_closed(self):
         if not self.open:
             raise Error("Attempting operation on closed cursor")
 
+    def _handle_staging_operation(self):
+        """Make HTTP request using instructions provided by server"""
+
+        row = self.active_result_set.fetchone()
+
+        # TODO: Handle headers. What format will gateway send? json? plaintext?
+        operation, presigned_url, local_file, headers = (
+            row.operation,
+            row.presignedUrl,
+            row.localFile,
+            None,
```

Comment:
Why are headers always None here? Did we parse the headers from the spark driver response?

Reply:
headers could be non-empty from Azure Cloud, please make sure we take care of this and considering integration tests that could run across cloud.

Reply:
The `None` here is a placeholder. I wrote above this block in the comments that I don't know what format the `headers` field will take. But to send them to `requests` I need to turn them into a Python `dict()`. To do that I need to know what format they will arrive in.

The sample notebook from @xiaonanyang-db doesn't include any values in the headers field that I saw. Does the Spark Driver send them as JSON?

Reply:
@susodapop got it. Yes, it would look like a JSON string like `{"h1": "v1"}`

Reply:
You can reference this test suite https://src.dev.databricks.com/databricks/runtime@afacc75d9c5436209fbac057d1f09caf9e30d196/-/blob/sql/hive-thriftserver/src/test/scala/com/databricks/sql/hive/thriftserver/StagingOperationSQLQuerySuite.scala?L65

Reply:
Interestingly, the response from the server today (with no headers) evaluates to an empty Python list `[]`. If it's supposed to be an empty JSON I would expect it to be `{}`. Which is I coerce it to starting in this commit: 55525cbf8660c7d47a628a2c072610b45c212ee8 (see line 315 in client.py)